### PR TITLE
[8.0.x] example: Add Semigroup

### PR DESCRIPTION
**Pull request by NeQuissimus**
Depends on #1546

### CODE REVIEW

File: example/src/main/tut/typeclass/Semigroup.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+layout: docs
+title:  "Semigroup"
+---
+
+# Semigroup
+
+**Category Theory**
+
+*A semigroup is an associative magma. It contains a set and an associative binary operation.*
```
Comment (hrhino) [inline example/src/main/tut/typeclass/Semigroup.md]: Mentioning magmas seems a little strange without a definition. A link to wiki/ncatlab may be welcome.

I'm not sure if it "contains" them; maybe
> ... magma. That is, it is a set along with an associative binary operation.

File: example/src/main/tut/typeclass/Semigroup.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+A semigroup instance is required to satisfy associativity such that the following holds true:
```
Comment (hrhino) [inline example/src/main/tut/typeclass/Semigroup.md]: "be associative" or "satisfy the associativity law"

File: example/src/main/tut/typeclass/Semigroup.md (hunk 1 above)
Comment (TomasMikula) [reply example/src/main/tut/typeclass/Semigroup.md]: Unless we have magma in scalaz, I would avoid mentioning it altogether.

File: example/src/main/tut/typeclass/Semigroup.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+---
+layout: docs
+title:  "Semigroup"
+---
+
+# Semigroup
+
+**Category Theory**
```
Comment (TomasMikula) [inline example/src/main/tut/typeclass/Semigroup.md]: Semigroup is not really from category theory.

**Comment by jdegoes** [comment]
🙏

**Pull request merged**